help-docs: Document "Private messages" mobile app feature.

drrosa opened the pull request:
Adds step-by-step instructions for mobile app users.

**Screenshots and screen captures:**

<img width="535" alt="image" src="https://user-images.githubusercontent.com/2343554/178348476-dccc378e-6d77-4a06-8709-8c675bfcec06.png">

**Comments:**

- I think it's reasonable to use the term "Conversations" because the iOS app uses it as the text label that refers to the tab that opens up the PMs view.

- The icon for the Private Messages tab in the mobile app is slightly different than the icon I was able to find i.e. 2 people vs 3 people, but I think it still makes sense to use it given that the icon for Group PMs is the same across platforms.

- The Send button doesn't have a text label in the mobile app, but I think it's reasonable to use it in the instructions without showing the corresponding icon as it would still make sense to the user.

---

Some preliminary work was needed for `Desktop/Web` in order to add consistent mobile documentation:

<img width="531" alt="image" src="https://user-images.githubusercontent.com/2343554/178348423-ce7ffc70-c19d-4393-8372-fe9467ed66ad.png">

<img width="727" alt="image" src="https://user-images.githubusercontent.com/2343554/178349131-0f3abc48-081e-4076-bbdc-6c53219af71a.png">

<details>
<summary>Self-review checklist</summary>

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

drrosa commented:
@alya ready for review.

alya commented on templates/zerver/help/private-messages.md:
```diff
@@ -9,38 +9,44 @@ frequently conversing with the same person or the same group of people, we
+1. Use the `x` keyboard shortcut to
+   [Open the compose box](/help/open-the-compose-box) or  
+   click **New private message** at the bottom of the screen.
+
```
Let's change this to:

> To start composing a private message, click the **New private message** button at the bottom of the Zulip window, or use the `x` keyboard shortcut.

I want to put the button UI before the power-user keyboard UI, and "bottom of the screen" doesn't seem right for web/desktop instructions, where the Zulip window might take up just part of the user's screen.

drrosa replied:
Sounds good! But I would suggest dropping "To start composing a private message, ". The instruction would be more concise, it'd be consistent with other articles, and also the last step begins with "Compose your message,".

drrosa replied:
And we can move the "compose box" link to the "Related articles" section.

alya commented on templates/zerver/help/private-messages.md:
```diff
@@ -9,38 +9,44 @@ frequently conversing with the same person or the same group of people, we
+   click **New private message** at the bottom of the screen.
+
+1. Start typing a user's name and select the user from the list of
    suggestions. Repeat as necessary.
```
Let's change this to something like:

> Start typing the name of the person you want to message, and select their name from the list of suggestions. You can continue adding as many message recipients as you like.

alya commented on templates/zerver/help/private-messages.md:
```diff
@@ -9,38 +9,44 @@ frequently conversing with the same person or the same group of people, we
 
-3. Compose a message, and click **Send**.
+1. Compose a message and click **Send**.
+
```
Let's change this to:

> Compose your message, and click **Send**.

alya commented on templates/zerver/help/private-messages.md:
```diff
@@ -9,38 +9,44 @@ frequently conversing with the same person or the same group of people, we
+
+    You can also click on a user's profile picture or name and then
+    click **Send private message**.
 
```
Nice addition! Maybe reword slightly?

> You can also click on a user's profile picture or name, and select **Send private message**.

alya commented:
> The Send button doesn't have a text label in the mobile app, but I think it's reasonable to use it in the instructions without showing the corresponding icon as it would still make sense to the user.

It's probably OK, but let's describe where to find it for clarity, i.e. "the **Send** button in the bottom right corner of the app."

drrosa commented:
I've pushed the requested changes!

If everything looks good, maybe we can merge this PR and we can add the icon for the PMs tab as a follow-up that also includes adding the ⓘ icon as discussed in #22435 and #22429.

[Private messages](https://zulip.com/help/private-messages)

<img width="527" alt="image" src="https://user-images.githubusercontent.com/2343554/179888342-7d705a15-22b0-4606-b89b-a1c42ed976c8.png">

<img width="529" alt="image" src="https://user-images.githubusercontent.com/2343554/179888385-35fabfdd-ce7b-4942-b5c7-39067114beb7.png">

<img width="202" alt="image" src="https://user-images.githubusercontent.com/2343554/179888663-2aa684d0-9446-42e1-ac13-83f9c364c150.png">

drrosa commented:
> Hm, where does the iOS app use it? Could you share a screenshot?

![iOS Conversations text label](https://user-images.githubusercontent.com/2343554/179893547-04eaa273-9809-4da9-905f-ea03c8c040a7.png)

alya commented:
> > Hm, where does the iOS app use it? Could you share a screenshot?
> 
> ![iOS Conversations text label](https://user-images.githubusercontent.com/2343554/179893547-04eaa273-9809-4da9-905f-ea03c8c040a7.png)

Ah, please take a look at https://chat.zulip.org/#narrow/stream/48-mobile/topic/platform.20differences/near/1391375 -- these labels are only shown in the tablet UI, which is a very small percentage of the user base for our mobile apps. The documentation should be written primarily with the phone UI in mind.

alya commented:
> If everything looks good, maybe we can merge this PR and we can add the icon for the PMs tab as a follow-up that also includes adding the ⓘ icon as discussed in https://github.com/zulip/zulip/pull/22435 and https://github.com/zulip/zulip/pull/22429.

The checkmark icon for the "Approve by.." instruction needs to be fixed as well, so I don't think we'll be able to merge this until the icons are figured out.

drrosa commented:
Made requested changes. Also added the location of the checkmark button in the app, and removed the icon for now.

<img width="528" alt="image" src="https://user-images.githubusercontent.com/2343554/180697066-638f949f-5d18-4c6a-a3f1-ffff903e23fd.png">

alya commented:
Hm, thinking about the actual instructions more carefully, those look correct for the group PM UI, but not for the 1:1 PM UI.

Perhaps we should have two separate sections, "Send a PM", and "Send a group PM"?

We could customize the desktop/web instructions a bit, too. I thin the tip about the user's profile picture or name makes sense for "Send a PM", but not for "Send a group PM".

What do you think?

drrosa commented:
> Hm, thinking about the actual instructions more carefully, those look correct for the group PM UI, but not for the 1:1 PM UI.
> 
> Perhaps we should have two separate sections, "Send a PM", and "Send a group PM"?
> 
> We could customize the desktop/web instructions a bit, too. I think the tip about the user's profile picture or name makes sense for "Send a PM", but not for "Send a group PM".
> 
> What do you think?

drrosa commented:
> Perhaps we should have two separate sections, "Send a PM", and "Send a group PM"? 
> What do you think?

That's a good question. Hm,  if we have two separate sections for 1:1 PM and group PM, only steps 2 and 4 would need to be different. Instead, I would suggest adding "If you are sending a group message..." to step 4 of the Mobile app instructions.

<img width="536" alt="image" src="https://user-images.githubusercontent.com/2343554/181649821-342246c1-abc0-4567-a9b2-3a9e48a07990.png">

> I think the tip about the user's profile picture or name makes sense for "Send a PM", but not for "Send a group PM".

That's true, but perhaps it could be a helpful tip to run into even if the user only wanted help with Group messaging.

alya commented:
Hm, I dunno, I'm still inclined to split it up. We should point out that you can also start a 1:1 PM on desktop/web from the right sidebar, but that instruction would make less sense for group PMs. The access instructions should ideally also be different for 1:1 vs. group PMs.

@laurynmm what do you think?

drrosa commented:
I think we've figured out a good strategy to add mobile icons, what do you think?

<img width="529" alt="image" src="https://user-images.githubusercontent.com/2343554/184791995-9e51b234-869a-45e9-8156-fc4abde9003a.png">

> Hm, I dunno, I'm still inclined to split it up. We should point out that you can also start a 1:1 PM on desktop/web from the right sidebar, but that instruction would make less sense for group PMs. The access instructions should ideally also be different for 1:1 vs. group PMs.
> 
> @laurynmm what do you think?

Also, I think that's a good point and perhaps it's best to split them up so I'll provide another update once I make the necessary changes.

drrosa commented:
@alya Pushed changes and updated screenshots ready for your review.

**Updates:**

- Splits instructions into "Send PM" / "Send group PM".
- Adds the corresponding image icons for the buttons that users should tap on their screen.
- Makes individual SVG image icons available in `/static/images/help/`, instead of importing the entire icon sets.
- Adds the icons for the "PMs" tab from https://feathericons.com.
- Adds the"Checkmark" and "Send" buttons from https://materialdesignicons.com.
- Adds a nested CSS selector to `.markdown img` so that the default white border for images does not get applied to the icons.

<img width="531" alt="image" src="https://user-images.githubusercontent.com/2343554/185943545-04c3149c-ce19-4832-a772-ebb734e75c8b.png">

<img width="528" alt="image" src="https://user-images.githubusercontent.com/2343554/185944143-9a6c6a7d-197f-49f3-8009-c12a1310d834.png">

alya commented:
Looks good to me! The icons are looking great. :)

@timabbott tagging for your review.

timabbott commented:
Merged, thanks @drrosa! One thing I notice while reading this is that probably the "`x` keyboard" shortcut part should be styled as a key, not an inline code block. May be worth trying to do a sweep for that sort of thing; in theory we want to be using HTML `<kbd>` elements.

@laurynmm FYI, since I remember you doing some cleanup of the keyboard logic.

laurynmm commented:
@drrosa - For context, this is the PR Tim's probably thinking about specifically: #22456. And I didn't really follow-up on [my last comment](https://github.com/zulip/zulip/pull/22456#issuecomment-1201397425) in that PR. Will do that today.
